app/assets/javascripts/talking_stick/talking_stick.js.erb

```diff
@@ -64,6 +65,67 @@ var TalkingStick = (function(self) {
       args = args[0];
     }
     self.localVideo.trigger(name, args);
+  };
+
+  self.detectDevices = function() {
+    if (typeof MediaStreamTrack === 'undefined' || typeof MediaStreamTrack.getSources === 'undefined') {
+      self.log('warning', 'MediaStreamTrack is not defined.  We will not be able to offer device selection.');
+    } else {
+      MediaStreamTrack.getSources(self.handleDevices);
+    }
+  };
+
+  self.handleDevices = function(devices) {
+    var microphones = 1;
+    var cameras = 1;
+    var tsDevices = [];
+    $.each(devices, function(i, device) {
```

Comment:
Maybe use [Array.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) which is supported in all real browsers and avoids an unnecessary jQuery invocation.